### DISCUSSION TITLE
TextMemoryPlugin + Azure AI Search

### DISCUSSION THREAD

**Discussion by yuichiromukaiyama** [microsoft/semantic-kernel#9341] [Q&A] [answered]
Is there a sample of using Azure AI Search (auth: api key) for memory in the [TextMemoryPlugin](https://github.com/microsoft/semantic-kernel/blob/2577cc9e51e20a78828a0ee9273e324275a13d19/python/semantic_kernel/core_plugins/text_memory_plugin.py#L21)...?

**Comment by markwallace-microsoft** [comment, marked as answer]
Hi @yuichiromukaiyama,
I recommend you take a look at our new Vector Store and Text Search abstractions as these are the features we plan to graduate.

**Samples**
[Memory Samples](https://github.com/microsoft/semantic-kernel/tree/main/dotnet/samples/Concepts/Memory)
[Consume From AzureAISearch](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Memory/VectorStore_ConsumeFromMemoryStore_AzureAISearch.cs)
[Vector Store AzureAISearch](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Memory/VectorStore_VectorSearch_MultiStore_AzureAISearch.cs)
[Search with Vector Store](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/GettingStartedWithTextSearch/Step4_Search_With_VectorStore.cs)
[Vector Store RAG](https://github.com/microsoft/semantic-kernel/tree/main/dotnet/samples/Demos/VectorStoreRAG)

**Documentation**
[Vector Store Connectors](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/?pivots=programming-language-csharp)
[Text Search](https://learn.microsoft.com/en-us/semantic-kernel/concepts/text-search/?pivots=programming-language-csharp)
